### PR TITLE
Update Go dependencies

### PR DESCRIPTION
**Pull request by bradbeck**
Direct dependencies are now defined in `dependencies.go` and have been updated with `go get -u`.

Likewise, generated CUE code for these dependencies has been updated as follows:
```bash
rm -rf cue.mod/gen
cue get go github.com/kyverno/kyverno/api/kyverno/v1
cue get go github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1
cue get go k8s.io/api/core/v1
cue get go k8s.io/api/rbac/v1
```

Signed-off-by: Brad Beck <bradley.beck@gmail.com>

### CODE REVIEW

File: tools.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package tools
```
Comment (sudo-bmitch) [inline tools.go:1]: I'm not a fan of "tools", feels like "util" and we could add other Go files here later.

```suggestion
package frsca
```

**Review by sudo-bmitch — CHANGES_REQUESTED** [review]

**Comment by sudo-bmitch** [comment]
If I run a `go mod tidy`, the above `cue` commands (we should have those in a script like we do for vendoring), I'm seeing the following in a `git diff`:

```
diff --git a/cue.mod/gen/github.com/kyverno/kyverno/api/kyverno/v1/image_verification_types_go_gen.cue b/cue.mod/gen/github.com/kyverno/kyverno/api/kyverno/v1/image_verification_types_go_gen.cue
index 89014b5..41dbd2e 100644
--- a/cue.mod/gen/github.com/kyverno/kyverno/api/kyverno/v1/image_verification_types_go_gen.cue
+++ b/cue.mod/gen/github.com/kyverno/kyverno/api/kyverno/v1/image_verification_types_go_gen.cue
@@ -67,17 +67,17 @@ import apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
        // Defaults to true.
        // +kubebuilder:default=true
        // +kubebuilder:validation:Optional
-       mutateDigest?: bool @go(MutateDigest)
+       mutateDigest: bool @go(MutateDigest)
 
        // VerifyDigest validates that images have a digest.
        // +kubebuilder:default=true
        // +kubebuilder:validation:Optional
-       verifyDigest?: bool @go(VerifyDigest)
+       verifyDigest: bool @go(VerifyDigest)
 
        // Required validates that images are verified i.e. have matched passed a signature or attestation check.
        // +kubebuilder:default=true
        // +kubebuilder:validation:Optional
-       required?: bool @go(Required)
+       required: bool @go(Required)
 }
 
 #AttestorSet: {
diff --git a/go.mod b/go.mod
index bc11d55..4cb6381 100644
--- a/go.mod
+++ b/go.mod
@@ -11,23 +11,18 @@ require (
 require (
        contrib.go.opencensus.io/exporter/ocagent v0.7.1-0.20200907061046-05415f1de66d // indirect
        contrib.go.opencensus.io/exporter/prometheus v0.4.1 // indirect
-       github.com/PuerkitoBio/purell v1.1.1 // indirect
-       github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578 // indirect
        github.com/beorn7/perks v1.0.1 // indirect
        github.com/blendle/zapdriver v1.3.1 // indirect
        github.com/census-instrumentation/opencensus-proto v0.3.0 // indirect
        github.com/cespare/xxhash/v2 v2.1.2 // indirect
        github.com/davecgh/go-spew v1.1.1 // indirect
-       github.com/emicklei/go-restful v2.16.0+incompatible // indirect
        github.com/emicklei/go-restful/v3 v3.9.0 // indirect
        github.com/evanphx/json-patch/v5 v5.6.0 // indirect
-       github.com/ghodss/yaml v1.0.1-0.20190212211648-25d852aebe32 // indirect
        github.com/go-kit/log v0.2.1 // indirect
        github.com/go-logfmt/logfmt v0.5.1 // indirect
        github.com/go-logr/logr v1.2.3 // indirect
        github.com/go-openapi/jsonpointer v0.19.5 // indirect
        github.com/go-openapi/jsonreference v0.20.0 // indirect
-       github.com/go-openapi/spec v0.20.6 // indirect
        github.com/go-openapi/swag v0.21.1 // indirect
        github.com/gogo/protobuf v1.3.2 // indirect
        github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da // indirect
@@ -36,11 +31,9 @@ require (
        github.com/google/go-cmp v0.5.8 // indirect
        github.com/google/go-containerregistry v0.11.0 // indirect
        github.com/google/gofuzz v1.2.0 // indirect
-       github.com/googleapis/gnostic v0.5.5 // indirect
        github.com/grpc-ecosystem/grpc-gateway v1.16.0 // indirect
        github.com/hashicorp/errwrap v1.1.0 // indirect
        github.com/hashicorp/go-multierror v1.1.1 // indirect
-       github.com/hashicorp/golang-lru v0.5.4 // indirect
        github.com/josharian/intern v1.0.0 // indirect
        github.com/json-iterator/go v1.1.12 // indirect
        github.com/kyverno/go-wildcard v1.0.4 // indirect
...
```

**Comment by bradbeck** [comment]
> If I run a `go mod tidy`, the above `cue` commands (we should have those in a script like we do for vendoring), I'm seeing the following in a `git diff`:
> 
The changes to `image_verification_types_go_gen.cue` are intentional. Those fields should be optional like the other fields in `#ImageVerification`. The comments on those fields say as much. Our existing usage breaks if these changes are not there.

The differences in `go.mod` are from using go v1.18 vs. v1.19 for the `go mod tidy`. I'll push the update `go.mod` since I now have go v1.19 installed locally.

**Review by sudo-bmitch — DISMISSED** [review]

**Review by sudo-bmitch — APPROVED** [review]

**Review by jondkent — APPROVED** [review]
Tests fine and LGTM

**Pull request merged**